utils/python/CIME/utils.py

```diff
@@ -872,3 +873,49 @@ def touch(fname):
         os.utime(fname, None)
     else:
         open(fname, 'a').close()
+
+def find_system_test(testname, case):
+    '''
+    Find and import the test matching testname
+    Look through the paths set in config_files.xml variable SYSTEM_TESTS_DIR
+    for components used in this case to find a test matching testname.  Add the
+    path to that directory to sys.path if its not there and return the test object
+    Fail if the test is not found in any of the paths.
+    '''
+    system_test_path = None
+    if testname in ("TESTRUNPASS", "TESTRUNDIFF", "TESTRUNFAIL", "TESTBUILDFAIL",
```

Comment:
done

On Thu, Jul 28, 2016 at 5:44 PM, James Foucar <notifications@github.com>
wrote:

> In utils/python/CIME/utils.py
> <https://github.com/ESMCI/cime/pull/304#discussion_r72722752>:
>
> > @@ -872,3 +873,49 @@ def touch(fname):
> >          os.utime(fname, None)
> >      else:
> >          open(fname, 'a').close()
> > +
> > +def find_system_test(testname, case):
> > +    '''
> > +    Find and import the test matching testname
> > +    Look through the paths set in config_files.xml variable SYSTEM_TESTS_DIR
> > +    for components used in this case to find a test matching testname.  Add the
> > +    path to that directory to sys.path if its not there and return the test object
> > +    Fail if the test is not found in any of the paths.
> > +    '''
> > +    system_test_path = None
> > +    if testname in ("TESTRUNPASS", "TESTRUNDIFF", "TESTRUNFAIL", "TESTBUILDFAIL",
>
> That would work as long as a comment was added somewhere in
> system_test_common.py that says to make test cases start with TEST.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ESMCI/cime/pull/304/files/4e6f7f5b9748dcacebd0754780647328c6356d5c#r72722752>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF16GImRawFfOUWdeLjsUDeuaQxL1D80ks5qaT77gaJpZM4JXp9->
> .
>



-- 
Jim Edwards

CESM Software Engineer
National Center for Atmospheric Research
Boulder, CO